### ISSUE TITLE
vault_identity_list defined but not used, linter does not detect all syntax error and not check all files

### ISSUE THREAD

**Issue by jb-graindorge-sc** [ansible/ansible-lint#3718]
##### Summary

In our main ansible repo there is our `ansible.cfg` file, containing a line `vault_identity_list` .
We excluded all the vaulted files in `.ansible-lint.yml` (because they only contains keys and we don't want to run linter in our Jenkins on those files).
When the`vault_identity_list` is defined in `ansible.cfg` but is not present in the filesystem ansible-lint just does syntax check and is not examining the whole playbook and the "sub" roles (it basically does not detect syntax error neither) 

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

tested on 6.17.2 and 6.18, ansible and ansible-lint installed with PIP

##### REPRODUCE

`vault_identity_list` defined in `ansible.cfg` but NOT PRESENT in filesystem
```
ansible-lint -c .ansible-lint.yaml my-playbook.yml
Passed: 0 failure(s), 0 warning(s) on 1 files. Last profile that met the validation criteria was 'production'.
```

`vault_identity_list` defined in `ansible.cfg` PRESENT in filesystem
```
ansible-lint -c .ansible-lint.yaml my-playbook.yml
Failed: 1 failure(s), 0 warning(s) on 12 files. Last profile that met the validation criteria was 'min'.
```
